packages/medusa/src/services/cart.ts

```diff
@@ -1735,77 +1756,140 @@ class CartService extends TransactionBaseService {
         )
 
         const { total, region } = cart
+
```

Comment:
self-review: I wanted to break down this method in smaller functions. Unfortunately I encounter a naming issue (in finding proper naming) as we have other session functions that would collide with the wanted naming. Therefore I keep that task for another time as I think we should have a separate service for cart payment session management.